From 3a2b861df5d4e4e3520188a9f24932a3251a0417 Mon Sep 17 00:00:00 2001
From: Jun Zhang <jun.zhang@arm.com>
Date: Tue, 15 Aug 2023 13:47:57 +0800
Subject: [PATCH] Increase MHU buffer and payload size for rss_comms

Fix some bugs and some changes for secure storage SP

Upstream-Status: Pending
Signed-off-by: Jun Zhang <jun.zhang@arm.com>
---
 components/rpc/rss_comms/caller/sp/rss_comms_protocol_embed.h | 2 +-
 platform/drivers/arm/mhu_driver/mhu_v3_x.h                    | 2 +-
 platform/drivers/arm/mhu_driver/mhu_wrapper_v3_x.c            | 1 -
 3 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/components/rpc/rss_comms/caller/sp/rss_comms_protocol_embed.h b/components/rpc/rss_comms/caller/sp/rss_comms_protocol_embed.h
index 1914143..efc987b 100644
--- a/components/rpc/rss_comms/caller/sp/rss_comms_protocol_embed.h
+++ b/components/rpc/rss_comms/caller/sp/rss_comms_protocol_embed.h
@@ -11,7 +11,7 @@
 #include <sys/cdefs.h>
 #include <psa/client.h>

-#define PLAT_RSS_COMMS_PAYLOAD_MAX_SIZE 0x500
+#define PLAT_RSS_COMMS_PAYLOAD_MAX_SIZE (0x40 + 0x1000)

 typedef struct psa_invec psa_invec;
 typedef struct psa_outvec psa_outvec;
diff --git a/platform/drivers/arm/mhu_driver/mhu_v3_x.h b/platform/drivers/arm/mhu_driver/mhu_v3_x.h
index 18bd910..ca1232e 100644
--- a/platform/drivers/arm/mhu_driver/mhu_v3_x.h
+++ b/platform/drivers/arm/mhu_driver/mhu_v3_x.h
@@ -63,7 +63,7 @@ extern "C" {
 #define MHU3_PBX_DBCH_FLAG_AP_COMMS     UINT32_C(0x2)

 /* Size of shared outband message buffer in bytes */
-#define MHU3_OUTBAND_BUF_SIZE UINT64_C(0x1000)
+#define MHU3_OUTBAND_BUF_SIZE UINT64_C(0x2000)
 #define MHU3_OUTBAND_BUF_HEADER_SIZE UINT64_C(0x4)

 /* Base of shared outband mesage buffer */
diff --git a/platform/drivers/arm/mhu_driver/mhu_wrapper_v3_x.c b/platform/drivers/arm/mhu_driver/mhu_wrapper_v3_x.c
index 60203bf..10ac84f 100644
--- a/platform/drivers/arm/mhu_driver/mhu_wrapper_v3_x.c
+++ b/platform/drivers/arm/mhu_driver/mhu_wrapper_v3_x.c
@@ -153,7 +153,6 @@ size_t mhu_get_max_message_size(void)
 static int validate_buffer_params(uintptr_t buf_addr, size_t buf_size)
 {
     if ((buf_addr == 0) || (!IS_ALIGNED(buf_addr, 4)) ||
-            (!IS_ALIGNED(buf_size, 4)) ||
             buf_size > mhu_get_max_message_size()) {
         return MHU_V_3_X_ERR_INVALID_PARAM;
     }
--
2.25.1

